### PR TITLE
fix(kit): `CalendarRange` not update `selectedActivePeriod`, when `value` updates

### LINKED ISSUE #8527: 🐞 - TuiCalendarRange change does not update checkbox position
### Playground Link

https://stackblitz.com/edit/angular-wm2zfa?file=src%2Fapp%2Fapp.component.ts

### Description

Hi!

When assigning a new value to the calendar's date field, the checkbox indicating the selected day does not update its position accordingly. The checkbox remains associated with the previous date instead of moving to the newly selected date.

When assigning a new value to the calendar's date field via a new object reference, the checkbox indicating the selected day does not update its position accordingly. The checkbox remains associated with the previous date instead of moving to the newly selected date. However, if the value is updated by selecting a key from the existing object (old reference), the issue does not occur, and the checkbox updates correctly.

### Angular version

17

### Taiga UI version

3.90

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by mdlufy**
Closes #8527

After update value:

**As Is:**

https://github.com/user-attachments/assets/8b6b4e83-8857-47e3-823a-47937be60391

**To Be:**

https://github.com/user-attachments/assets/a4c6a8f2-1a04-4b71-88b6-a55233c104c1

### CODE REVIEW

File: projects/kit/components/calendar-range/calendar-range.component.ts [hunk 1]
```diff
@@ -87,12 +87,23 @@ export class TuiCalendarRange implements OnInit, OnChanges {
         inject<Observable<TuiDayRange | null>>(TUI_CALENDAR_DATE_STREAM, {optional: true})
             ?.pipe(tuiWatch(this.cdr), takeUntilDestroyed())
             .subscribe((value) => {
+                if (value?.toString() !== this.selectedActivePeriod?.range?.toString()) {
+                    this.selectedActivePeriod = null;
+                }
+
```
Comment (splincode) [inline projects/kit/components/calendar-range/calendar-range.component.ts:93]: reuse in method

File: projects/kit/components/calendar-range/calendar-range.component.ts [hunk 2]
```diff
@@ -87,12 +87,23 @@ export class TuiCalendarRange implements OnInit, OnChanges {
...
-    public ngOnChanges(): void {
+    public ngOnChanges({value}: SimpleChanges): void {
         this.defaultViewedMonth = this.value?.from || this.defaultViewedMonth;
+
+        if (
+            (value?.currentValue as TuiDayRange)?.toString() !==
+            this.selectedActivePeriod?.range.toString()
+        ) {
+            this.selectedActivePeriod = null;
+        }
```
Comment (splincode) [inline projects/kit/components/calendar-range/calendar-range.component.ts:106]: reuse in method

File: projects/kit/components/calendar-range/calendar-range.component.ts (hunk 1 above)
Comment (mdlufy) [reply projects/kit/components/calendar-range/calendar-range.component.ts]: done

**Review by splincode — APPROVED** [review]

**Comment by waterplea** [comment]
Closing in favor of https://github.com/taiga-family/taiga-ui/pull/8595

**Pull request closed without merging**